src/ClientData/include/ClientData/CaptureData.h

```diff
@@ -246,6 +247,8 @@ class CaptureData {
 
   // Returns all the timers corresponding to scopes with non-invalid ids
   [[nodiscard]] std::vector<const TimerInfo*> GetAllScopeTimers(
+      absl::flat_hash_set<ScopeType> types = {ScopeType::kApiScope, ScopeType::kApiScopeAsync,
+                                              ScopeType::kDynamicallyInstrumentedFunction},
```

Comment:
I don't think we should add another parameter with a default value. This also leads to the situation where you might want to specify `min_tick` and `max_tick` but not the types and it's not possible.
I think you could have two methods: `GetAllScopeTimersForAllTypes` (or something like that, maybe "all" is not the desired word) and `GetAllScopeTimersForTypes` (or "for type" if you always only need one at a time) and one can use the other.
Otherwise you can pass the types every time if that use case is rare.